internal/stores/metadata.go

```diff
@@ -481,30 +489,51 @@ func (s *SQLStore) SetContractSet(ctx context.Context, name string, contractIds
 	return s.db.Model(&contractset).Association("Contracts").Replace(&dbContracts)
 }
 
-func (s *SQLStore) DeleteContractSet(ctx context.Context, name string) error {
-	return s.db.
-		Where(dbContractSet{Name: name}).
-		Delete(&dbContractSet{}).
-		Error
-}
-
 func (s *SQLStore) RemoveContract(ctx context.Context, id types.FileContractID) error {
```

Comment:
Can this be called `ArchiveContract` and just call `ArchiveContracts` internally?

Reply:
So originally I figured archiving and removing a contract is not the same thing. I didn't unify these methods because I figured we do not want to clear archived contracts from the known contracts list. Seeing as you made the suggestion to do that anyway, I unified all these methods. <https://github.com/SiaFoundation/renterd/pull/247/files#r1154287780>

(I thought it would pose a race condition when we process consensus changes for recently archived contracts, they would not be known and thus not be handled as such in any consensus changes processed after the archival)